p2p/providers/addrbook_provider/addrbook_provider.go

```diff
@@ -17,6 +17,7 @@ const ModuleName = "addrbook_provider"
 
 // AddrBookProvider is an interface that provides AddrBook accessors
 type AddrBookProvider interface {
+	modules.InitializableModule
```

Comment:
Nice catch 😉 but if you want to take it a step further you can just embed the `Module` interface that, in turn, implements these "3 musketeers"

Reply:
Oh, nice one - I hadn't considered that 😅